tests/unit/test_table_service.py

```diff
@@ -353,3 +353,64 @@ def test_drop_tables(mocker: MockerFixture, mock_table: Table):
     expected_query = "SELECT * FROM information_schema.tables WHERE table_name = ?"
     cursor_mock.execute.assert_any_call(expected_query, [ts.external_table_name])
     cursor_mock.execute.assert_any_call(expected_query, [ts.internal_table_name])
+
+
+def test_drop_outdated_partitions_table_without_partitions(
+    mocker: MockerFixture, mock_table: Table
+):
+    """
+    Test to check if the function 'drop_outdated_partitions' correctly
+    drops the partitions in the fact table that are outdated.
+    """
+
+    connection = MagicMock()
+    cursor_mock = MagicMock()
+    cursor_mock.execute.return_value = 0
+    connection.cursor.return_value = cursor_mock
+
+    ts = TableService(mock_table, connection)
+
+    with pytest.raises(FireboltError):
+        ts.drop_outdated_partitions()
+
+
+def test_drop_outdated_partitions(mocker: MockerFixture, mock_table_partitioned: Table):
+    """
+    Test to check if the function 'drop_outdated_partitions' correctly
+    drops the partitions in the fact table that are outdated.
+    """
+
+    connection = MagicMock()
+    cursor_mock = MagicMock()
+    connection.cursor.return_value = cursor_mock
+
+    does_table_exists_mock = mocker.patch(
+        "firebolt_ingest.table_service.does_table_exist", return_value=[True, True]
+    )
+
+    ts = TableService(mock_table_partitioned, connection)
+
+    cursor_mock.fetchall.return_value = [("user1", "12"), ("user2", "13")]
```

Comment:
Let's also add a test for when no outdated partitions are found, i.e. `cursor_mock.fetchall.return_value = []` and check that no drops were executed.
This way we ensure the operation is safe.

Reply:
added test_drop_outdated_partitions_no_found test